spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinderFactory.java

```diff
@@ -106,37 +110,59 @@ else if (!StringUtils.hasText(this.defaultBinder)) {
 					}
 				}
 				if (defaultCandidateConfigurations.size() == 1) {
-					this.defaultBinder = defaultCandidateConfigurations.iterator().next();
-					configurationName = this.defaultBinder;
+					this.defaultBindersPerBoundType.put(boundElementType.getName(),
+							defaultCandidateConfigurations.iterator().next());
```

Comment:
no, that's just the result of moving some code around and inlining. what you're suggesting makes more sense